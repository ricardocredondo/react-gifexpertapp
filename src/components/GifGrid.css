.gifgrid__titulo {
  padding: 0.3em 1.2em;
  font-size: 2.2rem;
  background-color: rgba(46, 139, 86, 0.514);
  border-radius: 3px;
}
.gifgrid_contenedor-imagen {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  align-content: space-between;
}
