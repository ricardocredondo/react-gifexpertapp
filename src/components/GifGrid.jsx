import React from 'react';
import PropTypes from 'prop-types';
import GifGridItem from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import './GifGrid.css';

const GifGrid = ({ categoria }) => {
  const { data, loading } = useFetchGifs(categoria);
  return (
    <section className="gifgrid__categoria">
      <h3 className="gifgrid__titulo">{categoria}</h3>
      {loading && <p>Loading</p>}
      <div className="gifgrid_contenedor-imagen">
        {data.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </section>
  );
};
GifGrid.propTypes = {
  categoria: PropTypes.string.isRequired,
};
export default GifGrid;
