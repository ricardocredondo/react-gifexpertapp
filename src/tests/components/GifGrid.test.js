import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');
describe('Pruebas en <GifGrid />', () => {
  const categoria = 'ramones';
  const mockUseFetchGifs = useFetchGifs;
  test('debe funcionar correctamente', () => {
    mockUseFetchGifs.mockImplementation((categoria) => ({
      data: [],
      loading: true,
    }));
    const wrapper = shallow(<GifGrid categoria={categoria} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
    const gifs = [
      {
        id: 'skdjf',
        title: 'la perra',
        url: 'http://localhost',
      },
      {
        id: 's89kdjf',
        title: 'la perra2',
        url: 'http://localhost',
      },
    ];
    mockUseFetchGifs.mockImplementation((categoria) => ({
      data: gifs,
      loading: false,
    }));
    const wrapper = shallow(<GifGrid categoria={categoria} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
