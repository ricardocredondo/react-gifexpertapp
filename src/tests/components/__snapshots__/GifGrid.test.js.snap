// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<section
  className="gifgrid__categoria"
>
  <h3
    className="gifgrid__titulo"
  >
    ramones
  </h3>
  <div
    className="gifgrid_contenedor-imagen"
  >
    <GifGridItem
      id="skdjf"
      key="skdjf"
      title="la perra"
      url="http://localhost"
    />
    <GifGridItem
      id="s89kdjf"
      key="s89kdjf"
      title="la perra2"
      url="http://localhost"
    />
  </div>
</section>
`;

exports[`Pruebas en <GifGrid /> debe funcionar correctamente 1`] = `
<section
  className="gifgrid__categoria"
>
  <h3
    className="gifgrid__titulo"
  >
    ramones
  </h3>
  <p>
    Loading
  </p>
  <div
    className="gifgrid_contenedor-imagen"
  />
</section>
`;
